### DISCUSSION TITLE
Prevent form completion

### DISCUSSION THREAD

**Discussion by watzon** [charmbracelet/huh#481] [Q&A]
Is there a way to prevent the form from completing when enter is pressed on the last field? I have a very specific use case that requires being able to explicitly submit the form and easily navigate between fields. I want to do something on each field update.

**Comment by jonas-grgt** [comment]
I had a similar issue and made this PR some days ago but haven't received any feedback yet: https://github.com/charmbracelet/huh/pull/469

As an alternative you could maybe add a confirmation field as last field?

**Reply by watzon** [reply]
That could possibly work. I also need to be able to call a function on every field update and I'm not seeing that either. As much as I love these libraries they may not be the way to go for what I want, or at the very least I may have to build out my own form components in bubbletea.
